sdk/include/mesh_dp_internal.h

```diff
@@ -0,0 +1,152 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Intel Corporation
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef __MESH_DP_INTERNAL_H
+#define __MESH_DP_INTERNAL_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "mesh_dp.h"
+#include "mcm_dp.h"
+#include <sys/queue.h>
+
+/**
+ * Mesh connection context structure
+ */
+typedef struct MeshConnectionContext {
+    /**
+     * NOTE: The __public structure is directly mapped in the memory to the
+     * MeshConnection structure, which is publicly accessible to the user.
+     * Therefore, the __public structure _MUST_ be placed first here.
+     */
+    MeshConnection __public;
+
+    /**
+     * NOTE: All declarations below this point are hidden from the user.
+     */
+
+    /**
+     * Connections list entry registered in the Mesh client.
+     */
+    LIST_ENTRY(MeshConnectionContext) conns;
+
+    /**
+     * MCM connection handle.
+     */
+    mcm_conn_context *handle;
+
+    /**
+     * Configuration structure
+     */
+    struct {
+        /**
+         * Connection kind (sender, receiver).
+         * Any value of the MESH_CONN_KIND_* constants.
+         */
+        int kind;
+
+        /**
+         * Connection type (memif, SMPTE ST2110-XX, RDMA).
+         * Any value of the MESH_CONN_TYPE_* constants.
+         */
+        int conn_type;
+#define MESH_CONN_TYPE_MEMIF  0 ///< Single node direct connection via memif
+#define MESH_CONN_TYPE_ST2110 1 ///< SMPTE ST2110-xx connection via Media Proxy
+#define MESH_CONN_TYPE_RDMA   2 ///< RDMA connection via Media Proxy
+
+        /**
+         * Configuration structures of all connection types
+         */
+        union {
+            MeshConfig_Memif memif;
+            MeshConfig_ST2110 st2110;
+            MeshConfig_RDMA rdma;
+        } conn;
+
+        /**
+         * Payload type (video, audio).
+         * Any value of the MESH_PAYLOAD_TYPE_* constants.
+         */
+        int payload_type;
+#define MESH_PAYLOAD_TYPE_VIDEO 0 ///< payload: video frames
+#define MESH_PAYLOAD_TYPE_AUDIO 1 ///< payload: audio packets
+
+        /**
+         * Configuration structures of all payload types
+         */
+        union {
+            MeshConfig_Video video;
+            MeshConfig_Audio audio;
+        } payload;
+    } cfg;
+
+} MeshConnectionContext;
+
+/**
+ * Heap zeroed allocation function definition
+ */
+#define mesh_calloc(size) calloc(1, size)
+
+/**
+ * Heap memory deallocation function definition
+ */
+#define mesh_free(ptr) free(ptr)
+
+/**
+ * Memory copy function definition
+ */
+#define mesh_memcpy(dst, src, size) memcpy(dst, src, size)
+
+/**
+ * String size-bounded copy function definition
+ */
+#define mesh_strlcpy(dst, src, size) strlcpy(dst, src, size)
```

Comment:
I would suggest not to introduce wrappers like mesh_calloc() etc. This can make confusion "why author of that code cannot use basic method"

Reply:
The idea behind these definitions was to create an isolation layer for future optimizations. I remember there were suggestions to use optimized memory operations related functions from the MTL repo.

Reply:
I agree with Tomek. Somebody will waste its time while reading the code to check what is underneath the wrapper, but there is nothing and plain calloc() could be used.

If there will be some optimalization underneath we can add the wrapper then

Reply:
Fixed in eae8b083e0210567994eb379b5e228742451c44b.